pkg/utils/k8sutil.go

```diff
@@ -383,6 +387,15 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 		dpm.Spec.Template.Spec.Containers[0].LivenessProbe = livenessProbe
 	}
 
+	// update env var for deployment
+	for _, env := range funcObj.Spec.Env {
+		k, v := getKV(env)
```

Comment:
*No!*
Using `getKV` here might pull environment values from the *controller* environment which is terrible - potentially exposing secrets.

This is why you do you "clever" string parsing asap on the client-side and just do dumb copying of datastructures everywhere downstream.